nexosim/src/server/services/inspector_service.rs

```diff
@@ -0,0 +1,583 @@
+    /// the registered sources.
+    pub(crate) fn get_event_source_schemas(
+        &self,
+        request: GetEventSourceSchemasRequest,
+    ) -> GetEventSourceSchemasReply {
+        match self {
+            Self::Started {
+                event_source_registry,
+                ..
+            } => {
+                let schemas = if request.source_names.is_empty() {
+                    // When no argument is provided return all the schemas.
+                    event_source_registry
+                        .list_sources()
+                        .map(|a| Ok((a.to_string(), event_source_registry.get_source_schema(a)?)))
+                        .collect()
+                } else {
+                    request
+                        .source_names
+                        .iter()
+                        .map(|a| Ok((a.to_string(), event_source_registry.get_source_schema(a)?)))
+                        .collect()
+                };
+
+                match schemas {
+                    Ok(schemas) => GetEventSourceSchemasReply {
+                        schemas,
+                        result: Some(get_event_source_schemas_reply::Result::Empty(())),
+                    },
+                    Err(e) => GetEventSourceSchemasReply {
+                        schemas: HashMap::new(),
+                        result: Some(get_event_source_schemas_reply::Result::Error(
+                            map_registry_error(e),
+                        )),
+                    },
+                }
+            }
+            Self::NotStarted => GetEventSourceSchemasReply {
+                schemas: HashMap::new(),
+                result: Some(get_event_source_schemas_reply::Result::Error(
+                    simulation_not_started_error(),
+                )),
+            },
+        }
+    }
+
+    /// Returns a list of names of all the registered query sources.
+    pub(crate) fn list_query_sources(&self, _: ListQuerySourcesRequest) -> ListQuerySourcesReply {
+        match self {
+            Self::Started {
+                query_source_registry,
+                ..
+            } => ListQuerySourcesReply {
+                source_names: query_source_registry
+                    .list_sources()
+                    .map(|a| a.to_string())
+                    .collect(),
+                result: Some(list_query_sources_reply::Result::Empty(())),
+            },
+            Self::NotStarted => ListQuerySourcesReply {
+                source_names: Vec::new(),
+                result: Some(list_query_sources_reply::Result::Error(
+                    simulation_not_started_error(),
+                )),
+            },
+        }
+    }
+
+    /// Retrieves the input and output query schemas for the specified query
+    /// sources.
+    ///
+    /// If the `source_names` field is left empty, it returns schemas for all
+    /// the registered sources.
+    pub(crate) fn get_query_source_schemas(
```

Comment:
The same here, I think it would be good to document that raw queries will still be listed but will have an empty string as schema